src/Intl/Idn/Idn.php

```diff
@@ -142,12 +142,13 @@ final class Idn
      * @param int    $options
      * @param int    $variant
      * @param array  $idna_info
+     * @param int    $phpVersion
      *
      * @return string|false
      */
-    public static function idn_to_ascii($domainName, $options = self::IDNA_DEFAULT, $variant = self::INTL_IDNA_VARIANT_UTS46, &$idna_info = [])
+    public static function idn_to_ascii($domainName, $options = self::IDNA_DEFAULT, $variant = self::INTL_IDNA_VARIANT_UTS46, &$idna_info = [], $phpVersion = \PHP_VERSION_ID)
     {
-        if (\PHP_VERSION_ID >= 70200 && self::INTL_IDNA_VARIANT_2003 === $variant) {
+        if ($phpVersion >= 70400 && self::INTL_IDNA_VARIANT_2003 === $variant) {
```

Comment:
`\PHP_VERSION_ID` should stay inline so that OPcache can perform dead code elimination.

Reply:
I can make changes, then code will be non testable.